dealership/vehicles.py

```diff
@@ -1,15 +1,30 @@
 class Vehicle(object):
     def __init__(self, maker, model, year, base_price, miles):
-        pass
-
+        self.maker = maker
+        self.model = model
+        self.year = year
+        self.base_price = base_price
+        self.miles = miles
 
 class Car(Vehicle):
-    pass
+    def sale_price(self):
+        return self.base_price * 1.2
+        
+    def purchase_price(self):
+        return self.sale_price() - (0.004 * self.miles)
```

Comment:
For each Vehicle subclass, you're repeating the sale_price and purchase_price functions, instead consider creating the function in the Vehicle class and defining the multipliers for each Vehicle subclass.